src/Services/LykkePay/LykkePayErc20DepositContractService.cs

```diff
@@ -0,0 +1,202 @@
+            }
+            catch (Exception e)
+            {
+                await _log.WriteErrorAsync(nameof(Erc20DepositContractService), nameof(CreateContract), "", e, DateTime.UtcNow);
+
+                return null;
+            }
+        }
+
+        public async Task<IEnumerable<string>> GetContractAddresses(IEnumerable<string> txHashes)
+        {
+            return await _contractService.GetContractsAddresses(txHashes);
+        }
+
+        public async Task<string> GetContractAddress(string userAddress)
+        {
+            var contract = await _contractRepository.Get(userAddress);
+
+            return contract?.ContractAddress;
+        }
+
+        public async Task ProcessAllAsync(Func<IErc20DepositContract, Task> processAction)
+        {
+            await _contractRepository.ProcessAllAsync(processAction);
+        }
+
+        public async Task<bool> ContainsAsync(string address)
+        {
+            var contains = await _contractRepository.Contains(address);
+
+            return contains;
+        }
+
+        /// <param name="depositContractAddress"></param>
+        /// <param name="erc20TokenAddress"></param>
+        /// <param name="destinationAddress"></param>
+        /// <returns>TransactionHash</returns>
+        public async Task<string> RecievePaymentFromDepositContract(string depositContractAddress,
+           string erc20TokenAddress, string destinationAddress)
+        {
+            var depositContract = await _contractRepository.GetByContractAddress(depositContractAddress);
+            if (depositContract == null)
+            {
+                throw new ClientSideException(ExceptionType.WrongParams, $"DepositContractAddress {depositContractAddress} does not exist");
+            }
+
+            var userWallet = await TransferWalletSharedService.GetUserTransferWalletAsync(_userTransferWalletRepository,
+                depositContractAddress, erc20TokenAddress, depositContract.UserAddress);
+
+            if (userWallet != null && !string.IsNullOrEmpty(userWallet.LastBalance))
+            {
+                throw new ClientSideException(ExceptionType.TransferInProcessing, $"Transfer from {depositContractAddress} was started before wait for it to complete");
+            }
+
+            var balance = await _ercInterfaceService.GetBalanceForExternalTokenAsync(depositContractAddress, erc20TokenAddress);
+            if (balance == 0)
+            {
+                throw new ClientSideException(ExceptionType.NotEnoughFunds, $"No tokens detected at deposit address {depositContractAddress}");
+            }
+
+            var guid = Guid.NewGuid();
+            var guidStr = guid.ToString();
+
+            var message = new Lykke.Service.EthereumCore.Core.Messages.LykkePay.LykkePayErc20TransferMessage()
+            {
+                OperationId = guidStr
+            };
+
+            var existingOperation = await _operationsRepository.GetAsync(guid.ToString());
```

Comment:
You have already defined guidStr variable earlier, use it.

Reply:
fixed